pkg/apis/events/v1beta1/cloudstoragesource_conversion.go

```diff
@@ -20,15 +20,46 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/google/knative-gcp/pkg/apis/convert"
+	v1 "github.com/google/knative-gcp/pkg/apis/events/v1"
 	"knative.dev/pkg/apis"
 )
 
 // ConvertTo implements apis.Convertible.
-func (*CloudStorageSource) ConvertTo(_ context.Context, to apis.Convertible) error {
-	return fmt.Errorf("v1beta1 is the highest known version, got: %T", to)
+// Converts source (from v1beta1.CloudStorageSource) into v1.CloudStorageSource.
+func (source *CloudStorageSource) ConvertTo(_ context.Context, to apis.Convertible) error {
+	switch sink := to.(type) {
+	case *v1.CloudStorageSource:
+		sink.ObjectMeta = source.ObjectMeta
+		sink.Spec.PubSubSpec = convert.ToV1PubSubSpec(source.Spec.PubSubSpec)
+		sink.Spec.ServiceAccountName = source.Spec.ServiceAccountName
+		sink.Spec.Bucket = source.Spec.Bucket
+		sink.Spec.EventTypes = source.Spec.EventTypes
+		sink.Spec.ObjectNamePrefix = source.Spec.ObjectNamePrefix
+		sink.Status.PubSubStatus = convert.ToV1PubSubStatus(source.Status.PubSubStatus)
+		sink.Status.NotificationID = source.Status.NotificationID
+		return nil
+	default:
+		return fmt.Errorf("unknown conversion, got: %T", sink)
```

Comment:
does this needs to convert as well?
As in https://github.com/knative/serving/blob/master/pkg/apis/serving/v1beta1/configuration_conversion.go#L35

Reply:
Same above, 
v1 is the highest version. So for v1beta1, it is ok that we return error here because v1beta1 only has one higher version v1.
Either way works.
Now, it is obvious to user that that v1 is the highest version.
If we change it to https://github.com/knative/serving/blob/master/pkg/apis/serving/v1beta1/configuration_conversion.go#L35, we can keep the code style consistent.
I'm fine with either.

Reply:
I also modified the code in v1beta1 to keep the code style consistent